custom_components/miele/translations/de.json

```diff
@@ -1,257 +1,436 @@
 {
-  "config": {
-    "abort": {
-      "already_configured": "Konto ist bereits konfiguriert",
-      "already_in_progress": "Konfigurationsfluss bereits im Gange",
-      "authorize_url_timeout": "Zeitüberschreitung beim Erzeugen der Autorisierungs-URL.",
-      "missing_configuration": "Die Komponente ist nicht konfiguriert. Bitte folge der Anleitung.",
-      "no_url_available": "Keine URL verfügbar. Für Informationen zu diesem Fehler, [überprüfe den Hilfe-Abschnitt]({docs_url})",
-      "oauth_error": "Ungültige Token-Daten erhalten.",
-      "reauth_successful": "Die Neuauthentifizierung war erfolgreich"
-    },
-    "create_entry": {
-      "default": "Erfolgreich authentifiziert"
-    },
-    "step": {
-      "pick_implementation": {
-        "title": "Authentifizierungsmethode auswählen"
-      },
-      "reauth_confirm": {
-        "title": "Neuauthentifizierung erforderlich",
-        "description": "Die Miele-Integration muss deinen Account neu authentifizieren: {account}"
-      }
-    }
-  },
-  "entity": {
-    "sensor": {
-      "status": {
-        "state": {
-          "off": "Aus",
-          "on": "Ein",
-          "programmed": "Programmiert",
-          "waiting_to_start": "Start verzögert",
-          "running": "In Betrieb",
-          "pause": "Pause",
-          "program_ended": "Ende",
-          "failure": "Fehler",
-          "program_interrupted": "Programm unterbrochen",
-          "idle": "Untätig",
-          "rinse_hold": "Spülstop",
-          "service": "Service",
-          "superfreezing": "SuperFrost",
-          "supercooling": "SuperKühlen",
-          "superheating": "Schnellaufheizen",
-          "supercooling_superfreezing": "SuperKühlen/SuperFrost",
-          "not_connected": "Nicht verbunden"
-        }
-      },
-      "program_type": {
-        "state": {
-          "normal_operation_mode": "Normalbetrieb",
-          "own_program": "Eigenes Programm",
-          "automatic_program": "Automatik-Programm",
-          "cleaning_care_program": "Reinigungs-/Pflege-Programm",
-          "maintenance_program": "Maintenance program"
+    "config": {
+        "abort": {
+            "already_configured": "Account is already configured",
```

Comment:
why is there a second, english entry?

Reply:
That was by mistake. New pull request is already there. It is not merged by now.

Reply:
> That was by mistake. New pull request is already there. It is not merged by now.

okay thanks.
just looking into this because i am uncertain if scripts in homeassistant will still work correctly when updating language files.

Reply:
@g3gg0 
> just looking into this because i am uncertain if scripts in homeassistant will still work correctly when updating language files.

Well, this is a potential problem. My research in this topic has come to following conclusions:
- In a new installation of the integration:
  - entity_id (the internal name) and the friendly name are constructed from the translated name according to the global language setting in HA (Settings->General->Language)

- In an existing installation the entity_id is what it was when created.  If you change the global language setting and restart HA only the friendly name will change to the new language. So automations and script will continue to work.
- If you remove the integration and add it back then scripts and automations need a review if language or translations  are changed.

- The language setting in personal profile just affects the translation of state values (On/Ein/På etc) and status attributes.